### PR TITLE
remove db connections from functions

### PR DESCRIPTION
**Pull request by rego21**
Issue: [#23](https://github.com/sousajf1/skill-network/issues/23)

- I followed the steps available at the following link: [Rocket: Databases](https://rocket.rs/v0.4/guide/state/#databases)
- I believe we can remove the "backend/database/mod.rs" file

### CODE REVIEW

**Comment by sousajf1** [comment]
Try to Run clippy localy 😄

**Comment by sousajf1** [comment]
Please rebase this

**Comment by sousajf1** [comment]
I've rebased this!

File: backend/Rocket.toml [hunk 1]
```diff
@@ -1,2 +1,5 @@
 [global]
-address="0.0.0.0"
\ No newline at end of file
+address="0.0.0.0"
+
+[global.databases]
+sqlite_logs = { url = "test.db" }
```
Comment (luizchagasjardim) [inline backend/Rocket.toml:5]: Let's rename this, since the database does not hold logs.

File: backend/src/api/employee.rs [hunk 2]
```diff
@@ -1,44 +1,49 @@
 use rocket::Rocket;
 use rocket_contrib::json::Json;
 
-use crate::database::establish_connection;
+//use crate::database::establish_connection;
```
Comment (luizchagasjardim) [inline backend/src/api/employee.rs:4]:
```suggestion
```

File: backend/src/api/employee_skill.rs [hunk 3]
```diff
@@ -1,49 +1,52 @@
 use rocket::Rocket;
 use rocket_contrib::json::Json;
 
-use crate::database::establish_connection;
+//use crate::database::establish_connection;
```
Comment (luizchagasjardim) [inline backend/src/api/employee_skill.rs:4]:
```suggestion
```

File: backend/src/api/employee_skill.rs [hunk 4]
```diff
@@ -1,49 +1,52 @@
...
 use crate::database::models::employee::Employee;
 use crate::database::models::employee_skill::{EmployeesSkill, NewEmployeesSkill};
 use crate::database::models::skill::Skill;
 use crate::result::BackendError;
-use diesel::SqliteConnection;
+use crate::LogsDbConn;
+//use diesel::SqliteConnection;
```
Comment (luizchagasjardim) [inline backend/src/api/employee_skill.rs:10]:
```suggestion
```

File: backend/src/api/skill.rs [hunk 5]
```diff
@@ -1,42 +1,41 @@
 use rocket::Rocket;
 use rocket_contrib::json::Json;
 
-use crate::database::establish_connection;
+//use crate::database::establish_connection;
```
Comment (luizchagasjardim) [inline backend/src/api/skill.rs:4]:
```suggestion
```

File: backend/src/main.rs [hunk 6]
```diff
@@ -12,13 +13,18 @@ use dotenv::dotenv;
 use result::BackendError;
 use rocket_cors::{AllowedHeaders, AllowedOrigins, Error};
 
+#[database("sqlite_logs")]
+pub struct LogsDbConn(pub diesel::SqliteConnection);
```
Comment (luizchagasjardim) [inline backend/src/main.rs:17]: I would change the name of this struct to just DbConn

**Review by luizchagasjardim — CHANGES_REQUESTED** [review]

**Review by sousajf1 — APPROVED** [review]
Lgtm!

**Review by luizchagasjardim — APPROVED** [review]
Very good!

**Pull request merged**